src/main/java/uk/gov/gchq/magmacore/database/MagmaCoreDatabase.java

```diff
@@ -97,4 +97,34 @@ public interface MagmaCoreDatabase {
      * @param out Output stream to dump to.
      */
     void dump(PrintStream out);
+
+    /**
+     * Begin a writeable transaction initially in READ mode,
+     * but in Jena it will switch to WRITE mode if updates are made.
+     *
+     * @return the {@link MagmaCoreDatabase}
+    */
+    MagmaCoreDatabase begin();
```

Comment:
The idea was to support method chaining, but this would also need the `void` methods to return `this` as well and it wouldn't work for the methods that already return a value, but something like:
```java
db.begin()
  .update(thing)
  .commit();
```
would have been possible. It depends on how the database is going to be used in practice and probably needs to evolve with experience. I can switch the methods to return `void` if you prefer.

Reply:
I didn't know you needed to return `this` in order to implement method chaining. Should we update the `create`, `update` and `delete` methods to do the same? Either that or I think we should leave them all as void for the time being and make it an issue to address in future.

Reply:
It's a leftover from some work I was doing with Another Collaborator and probably isn't required at present so I'll revert to `void` for now.